internal/adminapi/kong.go

```diff
@@ -64,33 +64,37 @@ func NewKongClientForWorkspace(
 	if err != nil {
 		return nil, fmt.Errorf("creating Kong client: %w", err)
 	}
-	if wsName == "" {
-		return NewClient(client), nil
-	}
-
 	// Ensure that the client is ready to be used by performing a status check.
 	if _, err := client.Status(ctx); err != nil {
 		return nil, KongClientNotReadyError{Err: err}
 	}
 
-	// If a workspace was provided, verify whether or not it exists.
-	exists, err := client.Workspaces.ExistsByName(ctx, kong.String(wsName))
-	if err != nil {
-		return nil, fmt.Errorf("looking up workspace: %w", err)
-	}
+	if wsName != "" {
+		// If a workspace was provided, verify whether or not it exists.
+		exists, err := client.Workspaces.ExistsByName(ctx, kong.String(wsName))
+		if err != nil {
+			return nil, fmt.Errorf("looking up workspace: %w", err)
+		}
 
-	// If the provided workspace does not exist, for convenience we create it.
-	if !exists {
-		workspace := kong.Workspace{
-			Name: kong.String(wsName),
+		// If the provided workspace does not exist, for convenience we create it.
+		if !exists {
+			workspace := kong.Workspace{
+				Name: kong.String(wsName),
+			}
+			if _, err := client.Workspaces.Create(ctx, &workspace); err != nil {
+				return nil, fmt.Errorf("creating workspace: %w", err)
+			}
 		}
-		if _, err := client.Workspaces.Create(ctx, &workspace); err != nil {
-			return nil, fmt.Errorf("creating workspace: %w", err)
+		// Ensure that we set the workspace appropriately.
+		client.SetWorkspace(wsName)
+
+		// Now that we have set the workspace, ensure that the client is ready
+		// to be used with said workspace.
+		if _, err := client.Status(ctx); err != nil {
+			return nil, KongClientNotReadyError{Err: err}
 		}
```

Comment:
Is this necessary, given we have the status checked already? I don't think the `/status` is workspace-aware, is it?

Reply:
I can't find the spec that would mention that status is workspace aware but what I can see in https://github.com/kong/go-kong/blob/v0.59.1/kong/client.go#L381 is that `Status()` calls [`NewRequest()`](https://github.com/kong/go-kong/blob/v0.59.1/kong/request.go#L65) which in turn uses [`c.workspacedBaseURL(c.Workspace())`](https://github.com/kong/go-kong/blob/v0.59.1/kong/request.go#L68C33-L68C67) so the request that goes out is indeed workspace aware.

I think we have 3 options: 
- leave as is: only call `Status()` before setting the workspace
- add the call to `Status()` after setting the workspace (what this PR does currently)
- call `Status()` only after the workspace is set

I'm leaning towards leaving this as proposed or checking the status only before the workspace is set (what worked prior to this PR).

Reply:
👍 Let's leave it as proposed.